core/domain/classifier_domain.py

```diff
@@ -80,3 +82,45 @@ def to_dict(self):
             'cached_classifier_data': self.cached_classifier_data,
             'data_schema_version': self.data_schema_version
         }
+
+    def validate(self):
+        """Validates the classifier before it is saved to storage."""
+
+        if not isinstance(self.id, basestring):
+            raise utils.ValidationError(
+                'Expected id to be a string, received %s' % self.id)
+        utils.require_valid_name(
```

Comment:
Also, I don't think this particular field needs to be validated as a human-readable "name". It's an algorithmically-generated string.